resources/aws_ec2.go

```diff
@@ -708,3 +718,30 @@ func (obj *AwsEc2Res) snsPostHandler(w http.ResponseWriter, req *http.Request) {
 		log.Printf(string(post))
 	}
 }
+
+// snsMakeTopic creates a topic on aws sns.
+func (obj *AwsEc2Res) snsMakeTopic() error {
+	// make sns client
+	snsSess, err := session.NewSession(&aws.Config{
+		Region: aws.String(obj.Region),
+	})
+	if err != nil {
+		return err
+	}
+	obj.snsClient = sns.New(snsSess)
```

Comment:
do we use snsClient somewhere later outside of this function? IOW why does it need to be in the struct instead of returning it or only having it as a local var?

Reply:
It's fine to do it this way as a side effect, just contrast with potentially returning it and then dealing with it that way instead. Both are valid, different advantages and disadvantages for both.